### ISSUE TITLE
Private repo dependencies act differently to public deps

### ISSUE THREAD

**Issue by kelostrada** [hexpm/hexpm#695]
I have dependencies tree like this:

```
my_project -> common_backoffice -> common
           -> common
```

I would like to edit `common` locally so I put a setting in `my_project` in mix deps like this:

```elixir
{:common, path: "../../../elixir_common_tools", override: true},
```

In `common_backoffice` mix deps is defined like this:

```elixir
{:common, "~> 1.1.1", organization: "xxx"},
```

And I get an error while doing `mix deps.get`

```
* Updating common_backoffice (Hex package)
Dependencies have diverged:
* common (../../../elixir_common_tools)
  different specs were given for the common app:

  > In apps/my_project/mix.exs:
    {:common, [env: :prod, path: "../../../elixir_common_tools", override: true, manager: :mix]}

  > In deps/common_backoffice/mix.exs:
    {:common, "~> 1.1.1", [env: :prod, hex: "common", repo: "hexpm:xxx", optional: false]}

  Ensure they match or specify one of the above in your deps and set "override: true"
** (Mix) Can't continue due to errors on dependencies
```

The error I get is: "Ensure they match or specify one of the above in your deps and set "override: true". Even though I've set `override: true`. I'm pretty sure that simmilar scenario works with public hex packages if for instance I wanted to override `plug` with my local copy I would still be able to run a `phoenix` project even though `phoenix` depends on `plug` also so the tree is simmilar:

```
my_project -> phoenix -> plug
           -> plug
```

**Comment by ericmj** [comment]
Can you include the output of `mix hex.info`?

I think the difference between your issue and the phoenix/plug example is that you are using an umbrella but the phoenix/plug example is pure dependencies. I don't think it has anything to do with private packages.

**Comment by ericmj** [comment]
@kelostrada Did you resolve the issue?

**Comment by kelostrada** [comment]
hey, I just use the relative paths in every project I have for now.

You were right though - I use umbrella. However my tree in umbrella looks kinda like this:

```
my_phoenix -> phoenix_common -> common
my_engine -> common
```

Where `my_phoenix` and `my_engine` are projects inside umbrella app. When I swap `common` to be used from my drive instead of hex then the issue happens. I think it doesn't matter if I switch it inside `my_phoenix` or `my_engine`. I can test some more with non-umbrella app later.

**Comment by ericmj** [comment]
It's a bit hard to follow because the names of the apps are changing. It would help if you show your whole dependency tree including umbrella children with `mix deps.tree`.

**Comment by kelostrada** [comment]
Oh sorry, I will post that soon

**Comment by kelostrada** [comment]
```
==> project
project
└── common (../../../elixir_common_tools) *override*
    └── phoenix ~> 1.3 (...)
==> project_web
project_web
├── project (../project)
│   └── common (../../../elixir_common_tools) *override*
│       └── phoenix ~> 1.3 (Hex package)
└── common_web ~> 1.1.2 (Hex package)
    └── common ~> 1.1.1 (Hex package)
==> backoffice
backoffice
├── project (../project)
│   ├── common (../../../elixir_common_tools) *override*
│   │   └── phoenix ~> 1.3 (Hex package)
└── common_backoffice ~> 1.1.6 (Hex package)
    ├── common ~> 1.1.1 (Hex package)
    └── phoenix ~> 1.3.2 (Hex package)
```

**Comment by ericmj** [comment]
You need to override `common` in `project_web` and/or `backoffice`.

The reason you can't override in `project` is because it is on the same level as `common_backoffice` where it is not overridden.

**Comment by kelostrada** [comment]
You were right, thanks! However I have to override it in all three which is a bit annoying, but not as much as overriding it in the dependent projects (like in common_backoffice or common_web). 

When I just override it in backoffice and/or project_web (but not in project) then it still throws the same error. 

Thanks for help
